File: ProyectosCSharp/SoftProg/SoftProgPersistance/GestClientes/DAO/ClienteDAO.cs
using SoftProgModel.GestClientes;
using SoftProgModel.RRHH;
using SoftProgPersistance.DAO;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgPersistance.GestClientes.DAO
{
    public interface ClienteDAO : IDAO<Cliente>
    {
        BindingList<Cliente> listarPorDNIoNombre(String DNIoNombre);
     }
}

File: BackJava_FrontCSharp_Ejemplos_SOAP_REST/BackJava/SoftProgDA/src/main/java/pe/edu/pucp/softprog/gestclientes/dao/ClienteDAO.java
package pe.edu.pucp.softprog.gestclientes.dao;

import java.util.ArrayList;
import pe.edu.pucp.softprog.dao.IDAO;
import pe.edu.pucp.softprog.gestclientes.model.Cliente;

public interface ClienteDAO extends IDAO<Cliente>{
    ArrayList<Cliente> listarPorDNIoNombre(String DNInombre);
}
